src/Altinn.Notifications.Email.Integrations/Producers/EmailSendingAcceptedProducer.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Altinn.Notifications.Email.Integrations.Configuration;
+
+using Confluent.Kafka;
+using Confluent.Kafka.Admin;
+
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.Notifications.Email.Integrations.Producers;
+
+/// <summary>
+/// Implementation of a Kafka producer
+/// </summary>
+public sealed class EmailSendingAcceptedProducer : IEmailSendingAcceptedProducer, IDisposable
+{
+    private readonly IProducer<Null, string> _producer;
+    private readonly KafkaSettings _kafkaSettings;
+    private readonly SharedClientConfig _sharedClientConfig;
+    private readonly ILogger<EmailSendingAcceptedProducer> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EmailSendingAcceptedProducer"/> class.
+    /// </summary>
+    public EmailSendingAcceptedProducer(KafkaSettings kafkaSettings, ILogger<EmailSendingAcceptedProducer> logger)
+    {
+        _kafkaSettings = kafkaSettings;
+        _logger = logger;
+        
+        _sharedClientConfig = new SharedClientConfig(kafkaSettings);
+
+        var config = new ProducerConfig(_sharedClientConfig.ClientConfig)
+        {
+            Acks = Acks.All,
+            EnableDeliveryReports = true,
+            EnableIdempotence = true,
+            MessageSendMaxRetries = 3,
+            RetryBackoffMs = 1000
+        };
+
+        _producer = new ProducerBuilder<Null, string>(config).Build();
+        EnsureTopicsExist();
+    }
+
+    /// <inheritdoc/>
+    public async Task<bool> ProduceAsync(string message)
+    {
```

Comment:
might need to consider a generic producer here or altering the logic for ensuring that a topic exists. If we get multiple producer's we will have multiple admin clients starting up at the same time competing to create topics. 

Given the cost of initializing and running a producer we should consider creating a single generic one.